src/pybind/mgr/influx/module.py

```diff
@@ -192,53 +313,61 @@ def send_to_influx(self):
-                              "manually and grant write privs to user "
-                              "'%s'", self.config['database'],
-                              self.config['username'])
-                client.create_database(self.config['database'])
-            else:
+        df_stats = self.get_df_stats()
+        daemon_stats = self.get_daemon_stats()
+        pg_summary = self.get_pg_summary(df_stats[1])
+
+        for client,conf in self.clients:
+            # using influx client get_list_database requires admin privs,
+            # instead we'll catch the not found exception and inform the user if
+            # db can not be created
+            try:
+                client.write_points(df_stats[0], 'ms')
+                client.write_points(daemon_stats, 'ms')
+                client.write_points(self.get_pg_summary(df_stats[1]))
+                self.set_health_checks(dict())
+            except ConnectionError as e:
+                # InfluxDBClient also has get_host and get_port but since we have the config here anyways...
+                self.log.exception("Failed to connect to Influx host %s:%d",
+                                   conf['hostname'], conf['port'])
+                self.set_health_checks({
+                    'MGR_INFLUX_SEND_FAILED': {
+                        'severity': 'warning',
+                        'summary': 'Failed to send data to InfluxDB server at %s:%d'
+                                   ' due to a connection error'
+                                   %(conf['hostname'], conf['port']),
+                        'detail': [str(e)]
+                    }
+                })
+            except ReadTimeout as e:
+                self.log.exception("Timeout waiting for response from influx host %s:%d",
+                                   conf['hostname'], conf['port'])
                 self.set_health_checks({
                     'MGR_INFLUX_SEND_FAILED': {
                         'severity': 'warning',
-                        'summary': 'Failed to send data to InfluxDB',
+                        'summary': 'Failed to send data to InfluxDB server at %s:%d'
+                                   ' due to a response timeout from server'
+                                   %(conf['hostname'], conf['port']),
                         'detail': [str(e)]
                     }
                 })
-                raise
+            except InfluxDBClientError as e:
+                if e.code == 404:
+                    self.log.info("Database '%s' not found, trying to create "
+                                  "(requires admin privs).  You can also create "
+                                  "manually and grant write privs to user "
+                                  "'%s'", conf['database'],
+                                  conf['username'])
+                    client.create_database(conf['database'])
+                else:
+                    self.set_health_checks({
```

Comment:
This `set_health_checks` is probably redundant, because in the next line we're re-raising the exception, which will cause the module to die (and that will cause a health warning about the module having had an unhandled exception)